packages/bus-query-operation/lib/Bindings.ts

```diff
@@ -148,29 +153,93 @@ export function materializeOperation(
         };
       }
 
-      const variables = op.variables.filter(variable => !bindings.has(variable));
+      // Only include non-projected variables in the sub-bindings that will be passed down recursively.
+      // This will result in non-projected variables being replaced with their InitialBindings values.
+      let subBindings = bindingsFactory.fromBindings(bindings);
+      for (const binding of bindings) {
```

Comment:
Huh cool, so that's faster because we don't have to call bindings[0] later (although it's only a constant cost) and also just better for readability. Neat, I didn't think of that. I'm wondering if it's also faster because we don't have to put the whole binding in a variable. I guess not because putting something in a variable is just making a pointer which is constant regardless..